sdk/logical/identity.proto

```diff
@@ -36,7 +36,12 @@ message Alias {
 	// Name is the identifier of this identity in its authentication source
 	string name = 3;
 
-	// Metadata represents the custom data tied to this alias
+	// Metadata represents the custom data tied to this alias. Fields added
+	// to it should have a low rate of change (or no change) because each
+	// change incurs a storage write, so quickly-changing fields can have
+	// a significant performance impact at scale. See the SDK's
+	// "aliasmetadata" package for a helper that eases and standardizes
+	// using this safely.
```

Comment:
We observed large numbers of entities created with aliases that have unique names and no metadata in the alias as well (such as instance IDs).

Reply:
Thanks for pointing that out, I'll add a check that each piece of metadata being added has a value.

Reply:
This is addressed in https://github.com/hashicorp/vault/pull/8783/commits/556f563c5b53a74e58828cea8a92a2afd86a4d76

Reply:
Let me know if you're actually seeing _no metadata_ where you would expect it. If that's the case, could you open or link a GH issue regarding it? Debugging the values being passed in would be best as a separate PR.